pilot/pkg/networking/core/v1alpha3/route/route_cache.go

```diff
@@ -96,7 +97,12 @@ func (r *Cache) DependentConfigs() []model.ConfigHash {
 		}.HashCode())
 	}
 	for _, vs := range r.VirtualServices {
-		configs = append(configs, model.ConfigKey{Kind: kind.VirtualService, Name: vs.Name, Namespace: vs.Namespace}.HashCode())
+		if model.UseGatewaySemantics(vs) {
+			// add http/tcp/tls routes to dependent configs
+			configs = addInternalParents(configs, vs)
```

Comment:
This is not the right way for xds cache to be aware of gateway API. Except for gateway controller, other istio modules should be unaware of K8s Gateway/Ingress

Reply:
> Except for gateway controller, other istio modules should be unaware of K8s Gateway/Ingress

Ideally we should do this, but so far I don't see another way to clear cache on route updates

Reply:
I donot look into it, but this looks not good to me

Reply:
@hzxuzhonghu I am not sure any other possibility. A Gateway-API VirtualService is synthetic and has dependencies outside itself. We could make it not say "GatewaySemantics" in the function name, but I am not sure how we can avoid this check.

Thinking about it more -- doesn't the same issue apply to `Ingress`?

Reply:
> Thinking about it more -- doesn't the same issue apply to `Ingress`?

No, please see https://github.com/istio/istio/pull/40692#discussion_r957258617